python/python/lance/torch/async_dataset.py

```diff
@@ -0,0 +1,41 @@
+from multiprocessing import Process, Queue
+from typing import Callable
+
+from torch.utils.data import IterableDataset
+
+
+def _worker_ep(
+    dataset_creator: Callable[[], IterableDataset],
+    queue: Queue,
+):
+    dataset = dataset_creator()
+    while True:
+        for item in dataset:
+            queue.put(item)
+        queue.put(None)
+
+
+class AsyncDataset(IterableDataset):
```

Comment:
Lance dataset interanlly has buffer, and pytorch dataLoader has buffer, what is the difference here , comparing to the others.

Reply:
This is async loading. Loading from disk, and converting to tensor is a non trivial amount of time. This moves that latency to a different process so we can improve gpu utilization

Reply:
torch.data.utils.DataLoader is the same?

Reply:
Dataloader can't handle iteratable dataset. Rather, we don't have a sampler in the dataset to prevent duplicate data. https://pytorch.org/docs/stable/data.html#iterable-style-datasets

Reply:
we should switch over once we add multipleprocess sampling to our dataset implementation